### DISCUSSION TITLE
How UI work in koreader

### DISCUSSION THREAD

**Discussion by Monirzadeh** [koreader/koreader#11402] [Q&A] [answered]
Hi.
i try to work with koreader ui and i want when click specific menu item show an empty page with title "Title" for later add more item (bottom,etc) in that page i be confined with [this](http://koreader.rocks/doc/modules/) document
anybody can help and provide a simple code sample.
currently i can call my function like this
```
function MyPlugin:onShowMyPlugin()
logger.dbg("First page should run here")
end
```
is there any sample code in repository that i can't find

**Comment by Frenzie** [comment, marked as answer]
I find UI development easier with `./kodev wbuilder` for quicker iterations while prototyping.

There's (some) documentation for most widgets, but mostly I'd just look at the code of something that more or less does what you want already.
http://koreader.rocks/doc/modules/ui.widget.container.framecontainer.html
http://koreader.rocks/doc/modules/ui.widget.keyvaluepage.html

**Reply by Monirzadeh** [reply]
Thanks i can show ui with that two link
but i don't get how should i use `./kodev wbuilder`. when i run that it show a page with some size print on screen.
can i do more things with that?
i can't find documentation for `wbuilder`

**Reply by Frenzie** [reply]
It's just a place you can load some widgets into so you don't have to perform 10 actions to show it. It also shows some samples of what you can do.
https://github.com/koreader/koreader/blob/b832d43d985778e36841b7d08cf3db5436f26249/tools/wbuilder.lua
